R/compute_CYT.R

```diff
@@ -0,0 +1,66 @@
+#' Compute cytolytic activity score
+#'
+#' `compute_CYT` computes cytolytic activity score as the geometric mean of immune
+#' cytolytic genes (Rooney et al., 2015).
+#'
+#' @importFrom stats na.omit
+#'
+#' @param RNA_tpm numeric matrix with rows=genes and columns=samples
+#' @param verbose A logical value indicating whether to display informative messages
+#'
+#' @return numeric matrix with rows=samples and columns=cytolytic activity score
+#'
+#' @export
+#'
+#' @examples
+#' # Example: Mariathasan cohort (Mariathasan et al., Nature, 2018)
+#' if (!requireNamespace("BiocManager", quietly = TRUE))
+#'  install.packages("BiocManager")
+#'
+#' BiocManager::install(c("biomaRt",
+#'  "circlize",
+#'  "ComplexHeatmap",
+#'  "corrplot",
+#'  "DESeq2",
+#'  "dplyr",
+#'  "DT",
+#'  "edgeR",
+#'  "ggplot2",
+#'  "limma",
+#'  "lsmeans",
+#'  "reshape2",
+#'  "spatstat",
+#'  "survival",
+#'  "plyr"))
+#'
+#' install.packages("Downloads/IMvigor210CoreBiologies_1.0.0.tar.gz", repos = NULL)
+#' library(IMvigor210CoreBiologies)
+#'
+#' data(cds)
+#' mariathasan_data <- preprocess_mariathasan(cds)
+#' gene_tpm <- mariathasan_data$tpm
+#' rm(cds)
+#'
+#' CYT <- compute_CYT(RNA_tpm = gene_tpm)
+#' head(CYT)
+compute_CYT <- function(RNA_tpm,
+                        verbose = TRUE) {
+
+  # Literature signature
+  sig_read <- c("GZMA", "PRF1")
+  match_sig_read <- match(sig_read, rownames(RNA_tpm))
```

Comment:
As with the other signature - what's the plan here?

Reply:
Maybe it could be a good idea to merge first ars_signatures branch with ols_coderelatedfixes branch!

Reply:
Good point indeed.